Library/Homebrew/exceptions.rb

```diff
@@ -65,6 +65,25 @@ def to_s; <<-EOS.undent
   end
 end
 
+class TapFormulaAmbiguityError < RuntimeError
+  attr_reader :name, :paths, :formulae
+
+  def initialize name, paths
+    @name = name
+    @paths = paths
+    @formulae = paths.map do |path|
+      path.to_s =~ HOMEBREW_TAP_PATH_REGEX
+      "#{$1}/#{$2.sub("homebrew-", "")}/#{path.basename(".rb")}"
+    end
+
+    super <<-EOS.undent
+      Multiple formulae found: #{formulae.map { |f| "\n       * #{f}" }.join}
```

Comment:
Yeh, `multiple taps` makes sense given the current behaviour. I do wonder if we want to just treat core as another tap, though (perhaps in another PR)? e.g. with this PR if something is found in multiple taps then it produces this nice message but not if it's in a tap and core with the same name e.g. `omega` and `homebrew/games/omega`. In the medium-term we've got a GSoC student working on tap priorities and long-term I think we want to be able to use Homebrew without core formulae at all and use only a tap. Interested in your thoughts here.